### PR TITLE
Update libssh2 submodule to track back from https://github.com/libssh2/libssh2

### PR DESCRIPTION
**Pull request by crisidev**
This is the first time I do this, so I may be missing something..

1) Switched the gitmodule to https://github.com/libssh2/libssh2
2) Update libssh2-sys Cargo.toml version to 0.2.20
3) Update ssh2-rs Cargo.toml dep on sys crate to 0.2.20

Am I missing anything? The integration testing are running fine, so it "should" be good.

### CODE REVIEW

**Comment by crisidev** [comment]
Once this is merged, I'll release 0.9.0 and 0.2.20

File: .gitmodules [hunk 1]
```diff
@@ -1,4 +1,4 @@
 [submodule "libssh2-sys/libssh2"]
 	path = libssh2-sys/libssh2
-	url = https://github.com/wez/libssh2
-	branch = pr484
+	url = https://github.com/libssh2/libssh2
+	branch = master
```
Comment (wez) [inline .gitmodules:4]: I think this is right, but submodules can be tricky.  I'd expect to see the submodule changes as part of this PR (eg: take a look at an earlier commit: https://github.com/alexcrichton/ssh2-rs/commit/e8d4dc165fbcd3fceeec96db9d1e9be9b67bef7b)

It's worth re-syncing the git modules and making sure that you've checked out the appropriate commit from master in `libssh2-sys/libssh2`, then `git add libssh2-sys/libssh2` and include it in this PR.

Comment (crisidev) [reply .gitmodules:4]: I hate git submodules to be honest :cry:  I think I missed adding the submodule dir and committing also that. Let me see if I can make something closer to the other PR. Thanks for the feedback.

Comment (crisidev) [reply .gitmodules:4]: I think I did it..

**Review by wez — APPROVED** [review]
LGTM

**Comment by crisidev** [comment]
> LGTM

Thanks a lot. Quick question about releasing 0.9.0: should I do `cargo publish` first in the sys crate and then in the main one?

**Comment by wez** [comment]
Yep, publish the sys crate first!  I think it is safe to try to do it in the wrong order, so don't worry: `cargo publish` should error out if you try that because it depends on an unreleased sys version and will refuse to let you do it.

**Pull request merged**